### PR TITLE
Feat/reveal claim

### PR DESCRIPTION
**Pull request by yash431garg**
- Description: Chompy Loading Animation, Perform reveal+claim
- ClickUp links:https://app.clickup.com/t/86bzpmd54, https://app.clickup.com/t/86bzte9vt
- What are the acceptance criteria? What are the steps to test that this code is working?
1. Loading Animation until the code is rendered
2. Executing reveal and claim in one transaction and Ui updates related to this flow.

### CODE REVIEW

File: app/components/Bot/Bot.tsx [hunk 1]
```diff
@@ -238,6 +284,9 @@ export default function BotMiniApp() {
 
   useEffect(() => {
     setIsLoading(false);
+    if (isLoggedIn && !user?.emails[0]?.address && !user?.wallets[0]?.address) {
+      storeDynamicUser();
+    }
```
Comment (neel-ds) [inline app/components/Bot/Bot.tsx:289]: @yash431garg Please remove these lines to fix the infinite requests issue.

File: app/components/Bot/Bot.tsx [hunk 2]
```diff
@@ -87,27 +93,54 @@ export default function BotMiniApp() {
...
+        primaryWallet!.address,
+        blockhash,
+        totalRevealTokenAmount,
+      );
 
-      const tx = await genBonkBurnTx(primaryWallet!.address, blockhash, 10);
+      const burnTx = await signer.signAndSendTransaction(tx);
 
-      await signer.signAndSendTransaction(tx);
+      await handleBurnAndClaim(burnTx?.signature);
```
Comment (marvinmarnold) [inline app/components/Bot/Bot.tsx:141]: You are overwriting existing code. Won't this break the existing app which is still expecting burn and claim to be separate?

File: app/api/question/reveal/route.ts [hunk 3]
```diff
@@ -23,5 +35,280 @@ export async function GET(req: Request) {
     (question) =>
       question.answerCount && question.answerCount >= MINIMAL_ANSWER_COUNT,
   );
+
+  // const getQuestionIds = () => questionMinAnswerFilter.map((item) => item.id);
+
   return Response.json(questionMinAnswerFilter);
 }
+
+export async function POST(req: Request) {
```
Comment (marvinmarnold) [inline app/api/question/reveal/route.ts:44]: Very long method, would be nice to break it up and refactor.

**Review by marvinmarnold — APPROVED** [review]
Some minor comments, but let's merge so I can test everything.

File: app/components/Bot/Bot.tsx (hunk 2 above)
Comment (yash431garg) [reply app/components/Bot/Bot.tsx:141]: No, we have a sepereate function for the burn and claim to work togther.

File: app/api/question/reveal/route.ts (hunk 3 above)
Comment (yash431garg) [reply app/api/question/reveal/route.ts:44]: Sure, working on that now.

**Review by marvinmarnold — APPROVED** [review]

**Pull request merged**